JavaScript/JS-Program-Execution.md

```diff
@@ -0,0 +1,20 @@
+
+# Program Execution in JavaScript
+
+When we run a JS program, an execution context is created, which is further divided into two phases:
+        
+        1.  Memory Allocation Phase (Variable Environment)
+        2.  Code Execution Phase (Thread of Execution)
+
+## First Phase
+In the first phase, the entire program is skimmed line by line and memory is allocated to functions and variables. In case of variables, undefined is stored as a placeholder and in case of functions, the whole code of function is stored.
```

Comment:
- In the first phase, the entire program is skimmed line by line, and memory is allocated to functions and variables. In the case of variables, undefined is stored as a placeholder and in the case of functions, the whole code of the function is stored.
Some grammatical and punctuation mistakes need to be corrected